### ISSUE TITLE
[Bug]: [date 9.20]tke regression:  dn restart

### ISSUE THREAD

**Issue by heni02** [matrixorigin/matrixone#11860]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):nightly-d2a0d67
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

tke status:
<img width="880" alt="企业微信截图_5ec2bb97-1d10-413d-85b6-af0f298d7491" src="https://github.com/matrixorigin/matrixone/assets/113406637/f73f4ff8-40cf-4813-9b38-d6672942040b">

nightly-regression-dis-dn-0 crash log:
<img width="1418" alt="企业微信截图_113873df-d97a-413c-a69c-ec19bd9f4857" src="https://github.com/matrixorigin/matrixone/assets/113406637/dce3fcf6-354e-4467-9b26-4c7e77e976e0">
{"level":"ERROR","time":"2023/09/20 21:22:39.704915 +0000","caller":"txnimpl/antidepend.go:42","msg":"error: r-w conflict"}
{"level":"INFO","time":"2023/09/20 21:22:39.704973 +0000","caller":"txnimpl/table.go:296","msg":"[Start]","txn-start-ts":"1695244959656378737-1","operation":"transfer-deletes","operand":"BLK<272512-272514-98a5d725-57f6-11ee-9d25-ca5e8419b602-0-0>","phase":"Phase_Freeze"}
{"level":"INFO","time":"2023/09/20 21:22:39.705011 +0000","caller":"txnimpl/table.go:257","msg":"depth-0 bmsql_district transfer delete from blk-98a5d725-57f6-11ee-9d25-ca5e8419b602-0-0 row-1754 to blk-d2d05673-57fb-11ee-9d25-ca5e8419b602-0-0 row-1754"}
{"level":"INFO","time":"2023/09/20 21:22:39.705033 +0000","caller":"txnimpl/table.go:304","msg":"[End]","txn-start-ts":"1695244959656378737-1","operation":"transfer-deletes","operand":"BLK<272512-272514-98a5d725-57f6-11ee-9d25-ca5e8419b602-0-0>","phase":"Phase_Freeze"}
{"level":"ERROR","time":"2023/09/20 21:22:39.705084 +0000","caller":"txnimpl/antidepend.go:42","msg":"error: r-w conflict"}
{"level":"INFO","time":"2023/09/20 21:22:39.705110 +0000","caller":"txnimpl/table.go:296","msg":"[Start]","txn-start-ts":"1695244959656378737-1","operation":"transfer-deletes","operand":"BLK<272512-272517-fa295766-57f6-11ee-9d25-ca5e8419b602-0-0>","phase":"Phase_Freeze"}
{"level":"INFO","time":"2023/09/20 21:22:39.705144 +0000","caller":"txnimpl/table.go:257","msg":"depth-0 bmsql_warehouse transfer delete from blk-fa295766-57f6-11ee-9d25-ca5e8419b602-0-0 row-175 to blk-d2cf6b99-57fb-11ee-9d25-ca5e8419b602-0-0 row-175"}
{"level":"INFO","time":"2023/09/20 21:22:39.705172 +0000","caller":"txnimpl/table.go:304","msg":"[End]","txn-start-ts":"1695244959656378737-1","operation":"transfer-deletes","operand":"BLK<272512-272517-fa295766-57f6-11ee-9d25-ca5e8419b602-0-0>","phase":"Phase_Freeze"}
{"level":"ERROR","time":"2023/09/20 21:22:39.729776 +0000","caller":"txnimpl/antidepend.go:42","msg":"error: r-w conflict"}
{"level":"INFO","time":"2023/09/20 21:22:39.729830 +0000","caller":"txnimpl/table.go:296","msg":"[Start]","txn-start-ts":"1695244959694527453-1","operation":"transfer-deletes","operand":"BLK<272512-272514-98a5d725-57f6-11ee-9d25-ca5e8419b602-0-0>","phase":"Phase_Freeze"}
{"level":"INFO","time":"2023/09/20 21:22:39.729868 +0000","caller":"txnimpl/table.go:257","msg":"depth-0 bmsql_district transfer delete from blk-98a5d725-57f6-11ee-9d25-ca5e8419b602-0-0 row-4985 to blk-d2d05673-57fb-11ee-9d25-ca5e8419b602-0-0 row-4985"}
{"level":"INFO","time":"2023/09/20 21:22:39.729888 +0000","caller":"txnimpl/table.go:304","msg":"[End]","txn-start-ts":"1695244959694527453-1","operation":"transfer-deletes","operand":"BLK<272512-272514-98a5d725-57f6-11ee-9d25-ca5e8419b602-0-0>","phase":"Phase_Freeze"}
{"level":"ERROR","time":"2023/09/20 21:22:39.729924 +0000","caller":"txnimpl/antidepend.go:42","msg":"error: r-w conflict"}
{"level":"INFO","time":"2023/09/20 21:22:39.729948 +0000","caller":"txnimpl/table.go:296","msg":"[Start]","txn-start-ts":"1695244959694527453-1","operation":"transfer-deletes","operand":"BLK<272512-272517-fa295766-57f6-11ee-9d25-ca5e8419b602-0-0>","phase":"Phase_Freeze"}
{"level":"INFO","time":"2023/09/20 21:22:39.729969 +0000","caller":"txnimpl/table.go:257","msg":"depth-0 bmsql_warehouse transfer delete from blk-fa295766-57f6-11ee-9d25-ca5e8419b602-0-0 row-498 to blk-d2cf6b99-57fb-11ee-9d25-ca5e8419b602-0-0 row-498"}
{"level":"INFO","time":"2023/09/20 21:22:39.729983 +0000","caller":"txnimpl/table.go:304","msg":"[End]","txn-start-ts":"1695244959694527453-1","operation":"transfer-deletes","operand":"BLK<272512-272517-fa295766-57f6-11ee-9d25-ca5e8419b602-0-0>","phase":"Phase_Freeze"}
{"level":"ERROR","time":"2023/09/20 21:22:39.800785 +0000","caller":"moerr/error.go:552","msg":"error: internal error: convert go error to mo error context deadline exceeded"}
{"level":"ERROR","time":"2023/09/20 21:22:39.800849 +0000","caller":"db/db_holder.go:202","msg":"error: internal error: internal error: convert go error to mo error context deadline exceeded"}
{"level":"INFO","time":"2023/09/20 21:22:39.828454 +0000","caller":"motrace/syncer.go:89","msg":"Wait signal done."}

nightly-regression-dis-dn-0 log:(crash time :2023/09/21 05:22:39)
http://139.155.148.31:30088/explore?orgId=1&left=%7B%22datasource%22:%22P8E80F9AEF21F6940%22,%22queries%22:%5B%7B%22refId%22:%22A%22,%22expr%22:%22%7Bnamespace%3D%5C%22mo-nightly-regression-20230920%5C%22,%20pod%3D%5C%22nightly-regression-dis-dn-0%5C%22%7D%20%7C%3D%20%60%60%22,%22queryType%22:%22range%22,%22datasource%22:%7B%22type%22:%22loki%22,%22uid%22:%22P8E80F9AEF21F6940%22%7D,%22editorMode%22:%22builder%22%7D%5D,%22range%22:%7B%22from%22:%221695244920000%22,%22to%22:%221695244979000%22%7D%7D
<img width="2462" alt="企业微信截图_e635d3e4-5aab-42f9-af95-e9aeaebe1b3f" src="https://github.com/matrixorigin/matrixone/assets/113406637/b0d1c0d2-6743-488e-b3e4-04ee6adf0f39">
found fatal info , uncertain if it is the reason for restarting?

found dn not oom:
http://139.155.148.31:30088/d/6581e46e4e5c7ba40a07646395ef7b23/kubernetes-compute-resources-pod?orgId=1&var-datasource=default&var-cluster=&var-namespace=mo-nightly-regression-20230920&var-pod=nightly-regression-dis-dn-0&from=1695225859190&to=1695272056439
<img width="1907" alt="企业微信截图_30420aed-fd8f-4f98-a4a3-e2b73322177d" src="https://github.com/matrixorigin/matrixone/assets/113406637/eb058e83-0d41-485c-b913-02f14ca43e47">

**Comment by heni02** [comment]
@zhangxu19830126 jiangwei found fatal info log, maybe is dn restart reason, plz confirm

**Comment by LeftHandCold** [comment]
2023-09-21 05:22:36.800 | {"level":"FATAL","time":"2023/09/20 21:22:36.800194 +0000","name":"tn-service.lockservice","caller":"lockservice/rpc.go:229","msg":"missing request handler","uuid":"00000000-0000-0000-0000-100000000000","method":"Lock","stacktrace":"github.com/matrixorigin/matrixone/pkg/lockservice.(*server).onMessage\n\t/go/src/github.com/matrixorigin/matrixone/pkg/lockservice/rpc.go:229\ngithub.com/matrixorigin/matrixone/pkg/common/morpc.(*server).onMessage\n\t/go/src/github.com/matrixorigin/matrixone/pkg/common/morpc/server.go:241\ngithub.com/fagongzi/goetty/v2.(*server).doConnection\n\t/go/pkg/mod/github.com/matrixorigin/goetty/v2@v2.0.0-20230811091418-ff72333eabdc/application.go:381\ngithub.com/fagongzi/goetty/v2.(*server).doStart.func2.1\n\t/go/pkg/mod/github.com/matrixorigin/goetty/v2@v2.0.0-20230811091418-ff72333eabdc/application.go:344"}

**Comment by LeftHandCold** [comment]
The reason why DN restarts is because lockservice/rpc has FATAL.

**Comment by LeftHandCold** [comment]
<img width="1573" alt="image" src="https://github.com/matrixorigin/matrixone/assets/14086886/13482dd9-6b22-45da-9eab-af090ebd86f6">
After the fatal event, "cn service is not ready" appeared. During the period, dn restarted normally, but "cn service is not ready" was still displayed.

**Comment by LeftHandCold** [comment]
<img width="2357" alt="image" src="https://github.com/matrixorigin/matrixone/assets/14086886/0f2eb4fd-502e-4f31-9844-cc58c750f6c6">
cn在21:16:39 出现一个invalid connection，之后频繁出现error，5 6分钟后dn fatal重启，1分钟后dn重启成功，但是cn始终异常

**Comment by zhangxu19830126** [comment]
还没有看

**Comment by zhangxu19830126** [comment]
不知道为什么DN收到一个莫名其妙的request，让DN退出了。返回错误代替fatal，并且在错误信息里面加上来源地址，来跟踪这个问题。

**Comment by zhangxu19830126** [comment]
先fix dn重启问题。为什么收到其他的request，后续在定位。

**Comment by zhangxu19830126** [comment]
fixed panic by #11927

**Comment by heni02** [comment]
偶现问题，最近一周回归测试无出现dn重启现象，先关闭issue
